src/main/java/io/airlift/units/ThreadsCount.java

```diff
@@ -0,0 +1,195 @@
+        }
+        catch (NumberFormatException e) {
+            throw new IllegalArgumentException(format("Cannot parse value '%s' as float", value), e);
+        }
+    }
+
+    private static int parseInteger(String value)
+    {
+        try {
+            return Integer.parseInt(value);
+        }
+        catch (NumberFormatException e) {
+            throw new IllegalArgumentException(format("Cannot parse value '%s' as integer", value), e);
+        }
+    }
+
+    static int nextPowerOfTwo(int x)
+    {
+        if (x == 0) {
+            return 1;
+        }
+
+        --x;
+        x |= x >> 1;
+        x |= x >> 2;
+        x |= x >> 4;
+        x |= x >> 8;
+        return (x | x >> 16) + 1;
+    }
+
+    @Override
+    public int compareTo(ThreadsCount o)
+    {
+        return Integer.compare(threadsCount, o.threadsCount);
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        ThreadsCount that = (ThreadsCount) o;
+        return threadsCount == that.threadsCount;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return threadsCount;
+    }
+
+    @Override
+    public String toString()
+    {
+        if (threadsCount == 1) {
+            return "1 thread";
+        }
+        return format("%d threads", threadsCount);
+    }
+
+    static final class MachineInfo
+    {
+        private static final Path CPU_INFO_PATH = Paths.get("/proc/cpuinfo");
+
+        // cache physical processor count, so that it's not queried multiple times during tests
+        private static volatile int physicalProcessorCount = -1;
+
+        private MachineInfo() {}
+
+        public static int getAvailablePhysicalProcessorCount()
+        {
+            if (physicalProcessorCount != -1) {
+                return physicalProcessorCount;
+            }
+
+            String osArch = System.getProperty("os.arch");
+            // logical core count (including container cpu quota if there is any)
+            int availableProcessorCount = Runtime.getRuntime().availableProcessors();
+            int totalPhysicalProcessorCount;
+            if ("amd64".equals(osArch) || "x86_64".equals(osArch)) {
+                checkState(exists(CPU_INFO_PATH), "%s does not exist", CPU_INFO_PATH.toString());
+                try {
+                    totalPhysicalProcessorCount = toIntExact(readAllLines(CPU_INFO_PATH)
+                            .stream()
+                            .filter(line -> line.matches("^processor\\s+: \\d"))
+                            .count());
+                }
+                catch (IOException e) {
+                    throw new UncheckedIOException(e);
+                }
+            }
+            else {
+                // ARM platforms do not support hyper threading, therefore each logical processor is separate core
```

Comment:
This code might run on any Java platform, not just ARM, so let's remove this comment